black.py

```diff
@@ -629,7 +629,12 @@ def format_file_in_place(
     `mode` and `fast` options are passed to :func:`format_file_contents`.
     """
     if src.suffix == ".pyi":
-        mode = evolve(mode, is_pyi=True)
+        mode = FileMode(
+            mode.target_versions,
+            mode.line_length,
+            mode.string_normalization,
+            is_pyi=True,
+        )
```

Comment:
I think you can use `dataclasses.replace(mode, is_pyi=True)`, which is kind of more idiomatic.